native_client/client.cc

```diff
@@ -238,6 +255,64 @@ ProcessFile(ModelState* context, const char* path, bool show_times)
   }
 }
 
+std::vector<meta_word>
+WordsFromMetadata(Metadata* metadata)
+{
+  std::vector<meta_word> word_list;
+
+  std::string word = "";
+  float word_start_time = 0;
+
+  // Loop through each character
+  for (int i=0; i < metadata->num_items; i++) {
+    MetadataItem item = metadata->items[i];
+
+    if (strcmp(item.character," ") != 0) {
+      word.append(item.character);
+    }
+
+    // Word boundary is either a space or the last character in the array
+    if (strcmp(item.character," ") == 0 || i == metadata->num_items-1) {
```

Comment:
Maybe put more `(` `)`, and the second part if kinda tricky when you read it not carefully, so the `||` in the middle triggers attention :/

Reply:
Shall I split it across two lines?

Reply:
maybe it'd be easier to follow yes, also there's a missing space after `,`